Casbin/Enforcer.Internal.cs

```diff
@@ -82,32 +82,39 @@ private bool InternalEnforce<TRequest>(in EnforceContext context, in TRequest re
 
         if (context.View.HasPolicyParameter && session.HasNextPolicy)
         {
-            int policyIndex = 0;
-            while (scanner.GetNext(out IPolicyValues outValues))
+            try
             {
-                TPolicy policyValues = (TPolicy)outValues;
-                session.PolicyIndex = policyIndex;
-
-                HandleBeforeExpression(in context, ref session, in effectChain, in requestValues, policyValues);
-                session.ExpressionResult = expressionHandler.Invoke(in context, session.ExpressionString,
-                    in requestValues, in policyValues);
-
-                if (session.IsChainEffector)
-                {
-                    HandleExpressionResult(in context, ref session, ref effectChain, in requestValues, policyValues);
-                }
-                else
-                {
-                    HandleExpressionResult(in context, ref session, Effector, in requestValues, policyValues);
-                }
-
-                if (session.Determined)
+                int policyIndex = 0;
+                while (scanner.GetNext(out IPolicyValues outValues))
```

Comment:
The first scan will try to get the lock, maybe we can wait a few time here

Reply:
Ok, I will try this later.